### PR TITLE
Add release notes

### PR DESCRIPTION
**Pull request by cw-pratik**
Summary of the changes (in less than 80 chars)

 * Added the ablity to view release notes for packages.

### CODE REVIEW

File: src/BaGet.Protocol/Models/PackageMetadata.cs [hunk 1]
```diff
@@ -114,6 +114,12 @@ public class PackageMetadata
         [JsonProperty("title")]
         public string Title { get; set; }
 
+        /// <summary>
+        /// The package's release notes.
+        /// </summary>
+        [JsonProperty("releaseNotes")]
+        public string ReleaseNotes { get; set; }
```
Comment (loic-sharma) [inline src/BaGet.Protocol/Models/PackageMetadata.cs]: All the properties on `PackageMetadata` are part of the official NuGet protocol (see [this](https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#registration-page-object)).

Please move this to `BaGetPackageMetadata`, as that is where all of BaGet's custom additions to the protocol are placed. See: https://github.com/loic-sharma/BaGet/blob/46acdaae2b78d820e30ac9a84ca5e23700444bf9/src/BaGet.Core/Metadata/BaGetPackageMetadata.cs#L8

**Comment by loic-sharma** [comment]
Hi, thanks for opening this pull request! Would you mind also updating the Azure Table  package service? These places will need to be updated:

* The `PackageEntity` type:

https://github.com/loic-sharma/BaGet/blob/05bfc61e1ff86517d2e4571f12524275c2355a0d/src/BaGet.Azure/Table/PackageEntity.cs#L12-L46

* The `TableOperationBuilder.AddPackage` method:

https://github.com/loic-sharma/BaGet/blob/05bfc61e1ff86517d2e4571f12524275c2355a0d/src/BaGet.Azure/Table/TableOperationBuilder.cs#L20-L50

* The `TablePackageService.AsPackage` method:

https://github.com/loic-sharma/BaGet/blob/05bfc61e1ff86517d2e4571f12524275c2355a0d/src/BaGet.Azure/Table/TablePackageService.cs#L223-L251

Very nice work! :)

**Comment by cw-pratik** [comment]
Thanks for the review. Will update according to commets.

I wanted to ask you one thing. I deployed this version of Baget and replaced it with the previous release. The issue is that the Release notes did not get updated. I could see the Section in the UI, but without the actual notes. 

For new packages release notes are updated properly. 

What extra would I need to do so that while upgrading the release notes for old packages are also reflected.

**Comment by loic-sharma** [comment]
Sadly today there’s no way to “reprocess” already uploaded packages. I opened an issue to track this:  https://github.com/loic-sharma/BaGet/issues/447

Today, you’ll need to :

1. Enable package hard deletion (see this [documentation](https://loic-sharma.github.io/BaGet/configuration/#enable-package-hard-deletions))
1. Download the packages you’d like to reprocess
1. Delete the packages using [`dotnet NuGet delete`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-delete)
1. Reupload the packages

**Comment by loic-sharma** [comment]
Your changes look good to me, I’ll test this when I’m back home and then merge this in!

File: src/BaGet.Core/Metadata/BaGetPackageMetadata.cs [hunk 2]
```diff
@@ -24,5 +24,11 @@ public class BaGetPackageMetadata : PackageMetadata
 
         [JsonProperty("repositoryType")]
         public string RepositoryType { get; set; }
+
+        /// <summary>
+        /// The package's release notes.
+        /// </summary>
+        [JsonProperty("releaseNotes")]
+        public string ReleaseNotes { get; set; }
```
Comment (loic-sharma) [inline src/BaGet.Core/Metadata/BaGetPackageMetadata.cs]: Could you move this to after `PackageTypes` but before `RepositoryUrl` to keep these properties alphabetically sorted?

File: src/BaGet.UI/src/DisplayPackage/DisplayPackage.tsx [hunk 3]
```diff
@@ -247,6 +248,10 @@ class DisplayPackage extends React.Component<IDisplayPackageProps, IDisplayPacka
               <Dependents packageId={this.state.package.id} />
             </ExpandableSection>
 
+            <ExpandableSection title="Release Notes" expanded={false}>
```
Comment (loic-sharma) [inline src/BaGet.UI/src/DisplayPackage/DisplayPackage.tsx]: Most packages don't have release notes. Let's hide the section if the package is missing release notes:

```ts
{this.state.package.releaseNotes &&
  <ExpandableSection title="Release Notes" expanded={false}>
    <div className="package-release-notes" >{this.state.package.releaseNotes}</div>
  </ExpandableSection>
}
```

**Comment by loic-sharma** [comment]
Ok I tested this and it worked like a charm! I left two more minor comments, I'll merge this in once you've addressed them. Thanks for all the help!

**Comment by loic-sharma** [comment]
Thank you @cw-pratik for the excellent pull request!

**Pull request merged**